features/step_definitions/applications/keys_steps.rb

```diff
@@ -1,97 +1,99 @@
   stub_backend_get_keys
 end
 
-Given /^I care about application keys$/ do
+Given "I care about application keys" do
   unstub_backend_get_keys
 end
 
-Given /^(application "[^"]*") has the following keys:$/ do |application, table|
+Given "{application} has the following keys:" do |application, table|
   fake_application_keys(application, table.raw.map(&:first))
 end
 
-Given /^(application "[^"]*") has (\d+) keys$/ do |application, number|
+Given "{application} has {int} keys" do |application, number|
   fake_application_keys_count(application, number)
 end
 
-Given /^application "([^"]*)" has no keys$/ do |name|
+Given "application {string} has no keys" do |name|
   step %(application "#{name}" has 0 keys)
 end
 
-Given /^the application of (buyer "[^"]*") has the following keys:$/ do |buyer, table|
+Given "the application of {buyer} has the following keys:" do |buyer, table|
   fake_application_keys(buyer.bought_cinstance, table.raw.map(&:first))
 end
 
-Given /^the application of (buyer "[^"]*") has (\d+) keys$/ do |buyer, number|
+Given "the application of {buyer} has {int} keys" do |buyer, number|
   fake_application_keys_count(buyer.bought_cinstance, number)
 end
 
-Given /^the backend will create key "([^"]*)" for (application "[^"]*")$/ do |key, application|
+Given "the backend will create key {string} for {application}" do |key, application|
   stub_request(:post, backend_application_url(application, '/keys.xml'))
     .to_return(status: fake_status(201), body: %(<key value="#{key}"/>))
   fake_application_keys(application, [key])
 end
 
-Given /^the backend will create key "([^"]*)" for an application$/ do |key|
+Given "the backend will create key {string} for an application" do |key|
   stub_request(:post, %r{/applications/(.*)/keys.xml(.*)})
     .to_return(status: fake_status(201), body: %(<key value="#{key}"/>))
 
   stub_request(:get, %r{/applications/(.*)/keys.xml(.*)})
     .to_return(status: fake_status(200), body: %(<keys><key value="#{key}"</key>))
 end
 
-Given /^the backend will delete key "([^"]*)" for (application "[^"]*")$/ do |key, application|
+Given "the backend will delete key {string} for {application}" do |key, application|
   stub_request(:delete, backend_application_url(application, "/keys/#{key}.xml?provider_key=#{application.provider_account.api_key}&service_id=#{application.service.backend_id}"))
     .to_return(status: fake_status(200), body: '')
 end
 
-Given %r{^the backend will delete all keys for (application "[^"]*")$} do |application|
+Given "the backend will delete all keys for {application}" do |application|
   application.keys.each do |key|
-    step %{the backend will delete key "#{key}" for application "#{application.name}"}
+    step %(the backend will delete key "#{key}" for application "#{application.name}")
   end
 end
 
-When /^I (press|follow) "([^"]*)" for application key "([^"]*)"$/ do |action, label, key|
+When "I {word} {string} for application key {string}" do |action, label, key|
```

Comment:
:( how will I know that possible words are `press` and `follow`? It might be my lack of experience with cucumber expressions and I imagine that it will be annoying to figure that out. In this case it is not even in transformation but I'd have to check step `I #{action} "#{label}" within "#application_key_#{key}`.

btw in the past I needed to check where a particular step is used so I could run a `grep` in the sources with the step regular expression. Do you know any way to lookup where a step defined as a Cucumber expression is being used?

Reply:
same comment applies to multiple steps below